cpp/ql/test/library-tests/dataflow/fields/ir-flow.ql

```diff
@@ -41,6 +37,28 @@ class Conf extends DataFlow::Configuration {
   }
 }
 
-from DataFlow::PathNode src, DataFlow::PathNode sink, Conf conf
-where conf.hasFlowPath(src, sink)
-select sink, src, sink, sink + " flows from $@", src, src.toString()
+class IRFieldFlowTest extends InlineExpectationsTest {
+  IRFieldFlowTest() { this = "IRFieldFlowTest" }
+
+  override string getARelevantTag() { result = "ir" }
+
+  override predicate hasActualResult(Location location, string element, string tag, string value) {
+    exists(Node source, Node sink, Conf conf, int n |
+      tag = "ir" and
+      conf.hasFlow(source, sink) and
+      n = strictcount(Node otherSource | conf.hasFlow(otherSource, sink)) and
```

Comment:
Hey, this is cool! I've opened https://github.com/github/codeql-c-analysis-team/issues/76 so that we consider generalizing it.

Reply:
Oh, wait, that's not doing what I thought. It's differentiating results based on the source of flow, not on the column range of the result. That means we'll need to update the line numbers in the expectation as the tests change. This is _also_ a cool idea that we could support within the test framework, so I've updated the linked issue with a proposed solution that works for both cases.

Reply:
I agree that it's a bit fragile. Hopefully future modifications to the file will only append to it, and thus the annotations will not need to be changed.

Reply:
Yeah, I don't think there's a better way to do it today anyway, and it is an important thing to test. We can definitely live with that bit of fragility until we get eventually around to improving the test framework to make this easier, even if that isn't soon.